### PR TITLE
Delete deprecated reef-application endpoints

### PR DESCRIPTION
**Pull request by avalmas-programize**
Delete:
- GET /reef-applications/:id : Gets application based on a hashId and the uid
- PUT /reef-applications/:id : Update reefApplication based on a hashId

Also there was a functionality that allowed users to edit some reef column through the above PUT endpoint. This is no longer available and thus deleted.

### CODE REVIEW

**Review by ericboucher — APPROVED** [review]

**Comment by ericboucher** [comment]
@tleraj out of caution, let's make sure that this doesn't break anything in the frontend

File: packages/api/src/reef-applications/reef-application.spec.ts [hunk 1]
```diff
@@ -51,46 +51,6 @@ export const reefApplicationTests = () => {
...
-        'updatedAt',
-        'user',
-        'reef',
-        'uid',
-        'id',
-      ),
-    });
-  });
-
   it('PUT /:hashId/reefs/:id updates a reef application using reefId', async () => {
```
Comment (ericboucher) [inline packages/api/src/reef-applications/reef-application.spec.ts:54]: should we also delete this test? @avalmas-programize or @tleraj could you have another look at this spec file and see if other tests should be removed or updated?

Comment (tleraj) [reply packages/api/src/reef-applications/reef-application.spec.ts:54]: This endpoint is being used on the frontend, so I think we should keep it.

Comment (ericboucher) [reply packages/api/src/reef-applications/reef-application.spec.ts:54]: Then the question is, why is it being used in the frontend? There might be more frontend changes needed to fix #437

**Review by ericboucher — APPROVED** [review]

**Pull request merged**